qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -569,6 +570,20 @@ def _create_cliff_2q_layer_2():
 _NUM_LAYER_1 = 20
 _NUM_LAYER_2 = 16
 
+# Construct a dense multiplication table
+_CLIFFORD_LAYER_NUMS=[]
+for layer in [0,1,2]:
+    _CLIFFORD_LAYER_NUMS.append([])
+    for idx, qc in enumerate(_CLIFFORD_LAYER[layer]):
+        nn = []
+        for inst in qc:
+            qubits = tuple(qc.find_bit(q).index for q in inst.qubits)
+            rhs = _num_from_2q_gate(op=inst.operation, qubits=qubits)
+            nn.append(rhs)
+        _CLIFFORD_LAYER_NUMS[layer].append(tuple(nn))
+_valid_indices=np.unique(list(itertools.chain(*itertools.chain(*_CLIFFORD_LAYER_NUMS))))
+_clifford_num_to_dense = { idx: ii for ii, idx in enumerate(_valid_indices)} 
+_CLIFFORD_COMPOSE_2Q_DENSE = (_CLIFFORD_COMPOSE_2Q[:, _valid_indices]).toarray()
```

Comment:
Just for my memo. I'm surprised to know the custom dense matrix is faster than scipy.lil_matrix. If we used lil_matrix, the benchmark takes 18.76 sec, so about half of the improvement comes from the use of the custom dense matrix.